### PR TITLE
update wavesj & add compaction to compileScript

### PR DESCRIPTION
**Pull request by vladislavpetushkov**
add compaction param to Node.compileScript
add tests for Node.compileScript

### CODE REVIEW

File: src/main/java/im/mak/paddle/Node.java [hunk 1]
```diff
@@ -467,6 +467,11 @@ public ScriptInfo compileScript(String source) {
         return throwErrorOrGet(() -> super.compileScript(source));
     }
 
+    //@Override
```
Comment (msmolyakov) [inline src/main/java/im/mak/paddle/Node.java:470]: Why is this commented out?

File: src/test/java/im/mak/paddle/NodeTest.java [hunk 2]
```diff
@@ -60,4 +62,32 @@ void errorIfCantWaitNBlocks() {
...
 
+    @Nested
+    class CompileScript {
+        private final String INITIAL_SCRIPT = fromFile("wallet.ride");
+        private final String ERROR_SCRIPT = fromFile("wallet.ride").replace("i.payment", "v.payment");
+
+        @Test
+        void canCompileScript() {
+            ScriptInfo scriptInfo = node().compileScript(INITIAL_SCRIPT);
+            assertThat(scriptInfo.script()).isNotEqualTo("");
```
Comment (msmolyakov) [inline src/test/java/im/mak/paddle/NodeTest.java:73]: What's expected? Better to add an explicit expected base64 string, which will be different for the next test of compaction

**Review by msmolyakov — CHANGES_REQUESTED** [review]

File: src/main/java/im/mak/paddle/Node.java (hunk 1 above)
Comment (vladislavpetushkov) [reply src/main/java/im/mak/paddle/Node.java]: fix

File: src/test/java/im/mak/paddle/NodeTest.java (hunk 2 above)
Comment (vladislavpetushkov) [reply src/test/java/im/mak/paddle/NodeTest.java]: added compiled script comparison

**Review by msmolyakov — APPROVED** [review]

**Pull request merged**